drivers/fuel_gauge/bq27z746/bq27z746.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2023, ithinx GmbH
+ * Copyright (c) 2023, Tonies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_bq27z746
+
+#include "bq27z746.h"
+
+#include <zephyr/drivers/fuel_gauge.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ti_bq27z746);
+
+static int bq27z746_read16(const struct device *dev, uint8_t reg, uint16_t *value)
+{
+	uint8_t i2c_data[2];
+	const struct bq27z746_config *cfg = dev->config;
+	const int status = i2c_burst_read_dt(&cfg->i2c, reg, i2c_data, sizeof(i2c_data));
+
+	if (status < 0) {
+		LOG_ERR("Unable to read register");
+		return status;
+	}
+	*value = sys_get_le16(i2c_data);
+
+	return 0;
+}
+
+static int bq27z746_write16(const struct device *dev, uint8_t reg, uint16_t value)
+{
+	uint8_t buf[2];
+	const struct bq27z746_config *cfg = dev->config;
+
+	sys_put_le16(value, buf);
+
+	return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
```

Comment:
I recommend replacing i2c_burst_write_dt with i2c_write_dt instead, as i2c_burst_write_dt does not work correctly on all platforms (gory details here: https://github.com/zephyrproject-rtos/zephyr/pull/12588)

Reply:
Changed to `i2c_write_dt` as proposed :+1: